Demo/Demo/ViewModels/BaseViewModel.swift

```diff
@@ -126,6 +127,44 @@ class BaseViewModel: ObservableObject, PayPalWebCheckoutDelegate, CardDelegate {
         cardClient.approveOrder(request: cardRequest, context: context)
     }
 
+    func checkoutWithNativeClient(orderId: String?) async throws -> NativeCheckoutResult {
+        guard let orderId = orderId else {
+
+            return NativeCheckoutResult.error(CoreSDKError(code: 0, domain: "Order Id is null", errorDescription: "Order Id is null"))
+        }
+        let nativeCheckoutClient = try await getNativeCheckoutClient()
+        let paypalRequest = PayPalRequest(orderID: orderId)
+
+        class PaypalCheckoutDelegate: PayPalDelegate {
+
+            var nativeCheckoutResult: NativeCheckoutResult?
+            func paypal(_ payPalClient: PayPalClient, didFinishWithResult result: PayPalResult) {
+                nativeCheckoutResult = NativeCheckoutResult.approved(ApprovalResult(
+                    orderId: result.orderID, payerId: result.payerID
+                ))
+            }
+
+            func paypal(_ payPalClient: PayPalClient, didFinishWithError error: CoreSDKError) {
+                nativeCheckoutResult = NativeCheckoutResult.error(error)
+            }
+
+            func paypalDidCancel(_ payPalClient: PayPalClient) {
+                nativeCheckoutResult = NativeCheckoutResult.cancel
+            }
+        }
+
+        let callbackDelegate = PaypalCheckoutDelegate()
+        nativeCheckoutClient.delegate = callbackDelegate
+        DispatchQueue.main.async {
+            nativeCheckoutClient.start(request: paypalRequest)
+        }
+        guard let nativeCheckoutResult = callbackDelegate.nativeCheckoutResult
+        else {
+            throw CoreSDKError(code: 0, domain: "checkout", errorDescription: "didn't get result from checkout")
```

Comment:
Same comment here on errors, we would want to bubble up the error from the SDK vs a merchant creating their own using our CoreSDKError.